bot/exts/utilities/bookmark.py

```diff
@@ -23,6 +23,40 @@
 )
 
 
+class BookmarkTitleSelectionForm(discord.ui.Modal):
+    """
+    The form where a user can fill in a custom title for their bookmark & submit it.
+
+    This form is only available when the command is invoked from a context menu.
+    """
+
+    def __init__(
+            self,
+            message: discord.Message,
+            action_bookmark_function: Callable[[discord.TextChannel, discord.Member, discord.Message, str], None],
+    ):
+        super().__init__(timeout=1000, title="Name your bookmark")
+        self.message = message
+        self.action_bookmark = action_bookmark_function
+
+    bookmark_title = discord.ui.TextInput(
+        label="Choose a title for you bookmark",
```

Comment:
I'd like to be very clear that this is completely optional. Users would only know this was optional if they knew that required modal inputs had a little red asterisk, which is easy to miss and not something everyone knows about
```suggestion
        label="Choose a title for you bookmark (optional)",
```